src/pf-modal/pf-modal.component.js

```diff
@@ -0,0 +1,254 @@
+ * @prop {boolean} backdrop Indicating whether Clicking the backdrop could hide the modal or not
+ * @prop {boolean} keyboard Indicating whether clicking the escape key could hide the modal or not
+ * @prop {boolean} open Indicating whether or not the modal is opend
+ */
+
+export class PfModal extends HTMLElement {
+
+  /*
+   * Only attributes listed in the observedAttributes property will receive this callback
+   */
+  static get observedAttributes() {
+    return ['backdrop', 'keyboard', 'open'];
+  }
+
+  /*
+   * Get the show/hide state of modal
+   *
+   * @returns {boolean} True if modal is visible
+   */
+  get open () {
+    return this.hasAttribute('open');
+  }
+
+  /*
+   * Set flag indicating modal is shown
+   *
+   * @param {boolean} val True to set modal visible
+   */
+  set open (val) {
+    if (this.open !== val) {
+      if (val) {
+        this.setAttribute('open', '');
+      } else {
+        this._hideModal();
+      }
+    }
+  }
+
+  /*
+   * Get the backdrop setting
+   *
+   * @returns {boolean} True if users are allowed to hide modal on clicking backdrop
+   */
+  get backdrop () {
+    return this.hasAttribute('backdrop');
+  }
+
+  /*
+   * Set flag indicating clicking backdrop of modal could hide modal
+   *
+   * @param {boolean} val True to enable backdrop clicking
+   */
+  set backdrop (val) {
+    if (this.backdrop !== val) {
+      if (val) {
+        this.setAttribute('backdrop', '');
+      } else {
+        this.removeAttribute('backdrop');
+      }
+    }
+  }
+
+  /*
+   * Get the keyboard setting
+   *
+   * @returns {boolean} True if users are allowed to hide modal when escape key is pressed
+   */
+  get keyboard () {
+    return this.hasAttribute('keyboard');
+  }
+
+  /*
+   * Set flag indicating pressing escape key could hide modal
+   *
+   * @param {boolean} val True to enable escape key reaction
+   */
+  set keyboard (val) {
+    if (this.keyboard !== val) {
+      if (val) {
+        this.setAttribute('keyboard', '');
+      } else {
+        this.removeAttribute('keyboard');
+      }
+    }
+  }
+
+  /*
+   * Called when element's attribute value has changed
+   *
+   * @param {string} attrName The attribute name that has changed
+   * @param {string} oldValue The old attribute value
+   * @param {string} newValue The new attribute value
+   */
+  attributeChangedCallback (attrName, oldValue, newValue) {
+    // When the modal is shown, update keyboard/screen reader behavior.
+    if (attrName === 'open' && newValue !== null) {
+      this._showModal();
+    }
+    if (attrName === 'keyboard') {
+      if (newValue !== null) {
+        this._element.addEventListener('keydown', this._keydownHandler, false);
+      } else {
+        this._element.removeEventListener('keydown', this._keydownHandler, false);
+      }
+    }
+  }
+
+  /*
+   * An instance of the element is created or upgraded
+   * We build up the complete DOM tree of web components
+   */
+  constructor () {
```

Comment:
We aren't quite here yet with v1 constructors. Can you use the v0 attachedCallback for now? Would like to change all these new components in a new PR for #27